src/components/CourseCard/EmailSettingsModal.jsx

```diff
@@ -1,57 +1,138 @@
+       * see what happens when the promise is rejected, switch the comments.
+       */
+
+      // eslint-disable-next-line no-unused-vars
+      new Promise((resolve, reject) => {
+        setTimeout(resolve, 2000);
+        // setTimeout(() => reject(new Error('Network Error')), 2000);
+      })
+        .then(() => {
+          this.props.onClose(hasEmailsEnabled);
+        })
+        .catch((error) => {
+          this.setState({
+            isSubmitting: false,
+            error,
+          });
+        });
+    });
+  };
+
+  handleOnClose = () => {
+    this.setState({
+      isSubmitting: false,
+      isFormChanged: false,
+      error: null,
+    });
+    this.props.onClose();
+  };
+
+  render() {
+    const {
+      error, isFormChanged, hasEmailsEnabled, isSubmitting,
+    } = this.state;
+    const { title } = this.props;
+
+    return (
+      <>
+        <Modal
+          title={`Email Settings for ${title}`}
+          body={
+            <>
+              {error && (
+                <StatusAlert
+                  alertType="danger"
+                  dialog={
+                    <div className="d-flex">
+                      <div>
+                        <FontAwesomeIcon className="mr-3" icon={['fas', 'exclamation-triangle']} />
+                      </div>
+                      <div>
+                        An error occurred while saving your email settings. Please try again.
+                      </div>
+                    </div>
+                  }
+                  dismissible={false}
+                  open
+                />
+              )}
+              <div className="form-check">
+                <Input
+                  type="checkbox"
+                  id="email-settings"
+                  checked={hasEmailsEnabled}
+                  disabled={isSubmitting}
+                  onChange={this.handleEmailSettingsChange}
+                />
+                <label className="form-check-label ml-2" htmlFor="email-settings">
+                  Receive course emails such as reminders, schedule updates, and
+                  other critical announcements.
+                </label>
+              </div>
+            </>
+          }
+          onClose={this.handleOnClose}
+          buttons={[
+            {
+              label: (
+                <>
+                  {isSubmitting &&
+                    <FontAwesomeIcon className="mr-2" icon={['fas', 'spinner']} spin />
+                  }
+                  {isSubmitting ? 'Saving' : 'Save'}
+                  {' changes'}
+                  {isSubmitting && '...'}
+                </>
+              ),
+              buttonType: 'primary',
+              disabled: isSubmitting || !isFormChanged,
+              onClick: this.handleSaveButtonClick,
+            },
+          ]}
+          open
+        />
+      </>
+    );
+  }
+}
 
 EmailSettingsModal.propTypes = {
   title: PropTypes.string.isRequired,
-  submitting: PropTypes.bool,
```

Comment:
The logic around `submitting` is local to `EmailSettingsModal`'s state, so we aren't taking in a `submitting` prop now.